### PR TITLE
Disable video

### PR DESCRIPTION
**Pull request by jreidinger**
After testing with @lslezak new behavior is:

1) own video is black one
2) when I speak I do not see black screen but instead other person with video
3) on lada pc he see my picture if I speak ( probably reason is that video is not disabled, it is just "frozen" )

### CODE REVIEW

File: src/app/components/feed/feeds.factory.js [hunk 1]
```diff
@@ -198,6 +198,12 @@
               });
             }
           });
+          if (type === "video") {
+            var track = stream.getVideoTracks();
```
Comment (ancorgs) [inline src/app/components/feed/feeds.factory.js]: Nitpicking: the variable should be called `tracks` (plural)

File: src/app/components/videochat/jh-video-chat.directive.js [hunk 2]
```diff
@@ -122,7 +122,11 @@
           }
           // If current one is still speaking, there is no need to change
           if ( !(current && current.getSpeaking()) ) {
-            vm.highlight.current = FeedsService.speakingFeed() || current || FeedsService.findMain();
+            if (FeedsService.speakingFeed() && FeedsService.speakingFeed().getVideoSubscription()){
```
Comment (ancorgs) [inline src/app/components/videochat/jh-video-chat.directive.js]: You should not use `getVideoSubscription` here. That function only works for publisher feeds (i.e. yourself and your screen sharings). Use something like `someFeed.isEnabled("video")` instead. That function will always give you the right value, no matter if it's a publisher or subscriber feed.

Comment (ancorgs) [inline src/app/components/videochat/jh-video-chat.directive.js]: Although it's almost impossible, there is a chance than the two consecutive calls to `FeedService.speakingFeed()` return different feeds. Better use a variable, like
```
var speaking = FeedService.speakingFeed();
if (speaking && speaking.isEnabled("video")) {
```

File: src/app/components/videochat/jh-video-chat.directive.js [hunk 3]
```diff
@@ -122,7 +122,11 @@
           }
           // If current one is still speaking, there is no need to change
           if ( !(current && current.getSpeaking()) ) {
```
Comment (ancorgs) [inline src/app/components/videochat/jh-video-chat.directive.js]: I wonder, shouldn't we change if `current` disables his/her video while speaking? I think we should. So I would change this to
```javascript
// If current one is still speaking and has not disabled the video, there is no need to change
if ( !(current && current.getSpeaking() && current.isEnabled("video") )) {
```
What do you think?

Comment (jreidinger) [inline src/app/components/videochat/jh-video-chat.directive.js]: yes, make sense

Comment (jreidinger) [inline src/app/components/videochat/jh-video-chat.directive.js]: question is what happen as current is also Speaking, so it ends up with findMain, which in the end is publisher, as current cannot disable video ( from testing his video is still open and only last picture is frozen )

File: src/app/components/videochat/jh-video-chat.directive.js (hunk 2 above)
Comment (jreidinger) [inline src/app/components/videochat/jh-video-chat.directive.js]: ok

File: src/app/components/videochat/jh-video-chat.directive.js (hunk 3 above)
Comment (ancorgs) [inline src/app/components/videochat/jh-video-chat.directive.js]: Of course if everybody is disabling the video you will end up with some weird situation. We probably need to have placeholders also in the main feed view to handle that. But right now I would consider that a non relevant corner case.

**Comment by ancorgs** [comment]
lgtm

**Pull request merged**